pkg/skaffold/build/knative.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/knative"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/tag"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha2"
+	"github.com/pkg/errors"
+)
+
+// KnativeBuilder can build docker artifacts on Kubernetes, using knative build CRD.
+type KnativeBuilder struct {
+	*v1alpha2.KnativeBuild
+}
+
+// NewKnativeBuilder creates a KnativeBuilder.
+func NewKnativeBuilder(cfg *v1alpha2.KnativeBuild) *KnativeBuilder {
+	return &KnativeBuilder{
+		KnativeBuild: cfg,
+	}
+}
+
+// Labels gives labels to be set on artifacts deployed with knative.
+func (k *KnativeBuilder) Labels() map[string]string {
+	return map[string]string{
+		constants.Labels.Builder: "knative",
+	}
+}
+
+// Build builds a list of artifacts with knative.
+func (k *KnativeBuilder) Build(ctx context.Context, out io.Writer, tagger tag.Tagger, artifacts []*v1alpha2.Artifact) ([]Artifact, error) {
+	// TODO(dgageot): parallel builds
+	var builds []Artifact
+
+	for _, artifact := range artifacts {
+		fmt.Fprintf(out, "Building [%s]...\n", artifact.ImageName)
```

Comment:
what do you think about [outputting this in color (blue?)](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/color/formatter.go)?